internal/pool/pool.go

```diff
@@ -148,6 +124,42 @@ func (p *ConnPool) NewConn() (*Conn, error) {
 	return NewConn(netConn), nil
 }
 
+func (p *ConnPool) PopFree() *Conn {
+	timer := timers.Get().(*time.Timer)
+	if !timer.Reset(p.poolTimeout) {
+		<-timer.C
```

Comment:
This is needed, because if timer is already expired there will be 2 events in timer channel - http://play.golang.org/p/eq_bvcd0AB